engine/table/src/main/java/io/deephaven/engine/table/impl/select/CompareUtils.java

```diff
@@ -0,0 +1,147 @@
+//
+// Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.engine.table.impl.select;
+
+import io.deephaven.time.DateTimeUtils;
+import io.deephaven.util.compare.ObjectComparisons;
+import io.deephaven.util.type.NumericTypeUtils;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.time.Instant;
+
+class CompareUtils {
```

Comment:
![image](https://github.com/user-attachments/assets/2a599bf5-4ed2-4bb8-ab77-b62e991078d7)
We should try to do a little bit better for this function and make sure we have the edge cases covered.